dspace-api/src/main/java/org/dspace/service/impl/ClientInfoServiceImpl.java

```diff
@@ -74,40 +73,86 @@ public String getClientIp(String remoteIp, String xForwardedForHeaderValue) {
+     * which it usually uses to specify the original client IP address.
+     * @return IPTable of trusted IP addresses/ranges, or null if none could be found.
+     */
+    private IPTable parseTrustedProxyRanges() {
+        IPTable ipTable = null;
+        // Whether we had to look up the UI's IP address (based on its URL), or not
+        boolean uiUrlLookup = false;
+
+        String[] ipAddresses = configurationService.getArrayProperty("proxies.trusted.ipranges");
+        String uiUrl = configurationService.getProperty("dspace.ui.url");
+
+        // If configuration is empty, determine IPs of ${dspace.ui.url} as the default trusted proxy
+        if (ArrayUtils.isEmpty(ipAddresses)) {
+            // Get any IP address(es) associated with our UI
+            ipAddresses = Utils.getIPAddresses(uiUrl);
+            uiUrlLookup = true;
+        }
+
+        // Only continue if we have IPs to trust.
+        if (ArrayUtils.isNotEmpty(ipAddresses)) {
+            ipTable = new IPTable();
             try {
-                for (String proxyRange : proxyProperty) {
-                    ipTable.add(proxyRange);
+                // Load all IPs into our IP Table
+                for (String ipAddress : ipAddresses) {
+                    ipTable.add(ipAddress);
                 }
             } catch (IPTable.IPFormatException e) {
-                log.error("Property proxies.trusted.ipranges contains an invalid IP range", e);
+                if (uiUrlLookup) {
+                    log.error("IP address found for dspace.ui.url={} was invalid", uiUrl, e);
+                } else {
+                    log.error("Property 'proxies.trusted.ipranges' contains an invalid IP range", e);
+                }
                 ipTable = null;
             }
+        }
 
-            return ipTable;
+        if (ipTable != null) {
+            log.info("Trusted proxies (configure via 'proxies.trusted.ipranges'): {}", ipTable.toSet().toString());
         }
+
+        return ipTable;
     }
 
+    /**
+     * Whether a request is from a trusted proxy or not. Only returns true if trusted proxies are specified
+     * and the ipAddress is contained in those proxies. False in all other cases
+     * @param ipAddress IP address to check for
+     * @return true if trusted, false otherwise
+     */
     private boolean isRequestFromTrustedProxy(String ipAddress) {
         try {
-            return trustedProxies == null || trustedProxies.contains(ipAddress);
+            return trustedProxies != null && trustedProxies.contains(ipAddress);
         } catch (IPTable.IPFormatException e) {
             log.error("Request contains invalid remote address", e);
             return false;
         }
     }
 
+    /**
+     * Get the first X-FORWARDED-FOR header value which does not match the IP or another proxy IP. This is the most
+     * likely client IP address when proxies are in use.
+     * @param remoteIp remote IP address
```

Comment:
Could you also explicitly mention that this method doesn't verify whether it's a valid X-FORWARDED-FOR value (that it can still be a spoofed value)

Reply:
done